docs/Options.md

```diff
@@ -0,0 +1,530 @@
+| `SPACESHIP_TIME_12HR` | `false` | Format time using 12-hour clock (am/pm) |
+
+### Username (`user`)
+
+By default, a username is shown only when it's not the same as `$LOGNAME`, when you're connected via SSH or when you're root. Root user is highlighted in `SPACESHIP_USER_COLOR_ROOT` color (red as default).
+
+| Variable | Default | Meaning |
+| :------- | :-----: | ------- |
+| `SPACESHIP_USER_SHOW` | `true` | Show user section (`true`, `false`, `always` or `needed`) |
+| `SPACESHIP_USER_PREFIX` | `with ` | Prefix before user section |
+| `SPACESHIP_USER_SUFFIX` | `$SPACESHIP_PROMPT_DEFAULT_SUFFIX` | Suffix after user section |
+| `SPACESHIP_USER_COLOR` | `yellow` | Color of user section |
+| `SPACESHIP_USER_COLOR_ROOT` | `red` | Color of user section when it's root |
+
+`SPACESHIP_USER_SHOW` defines when to show username section. Here are possible values:
+
+| `SPACESHIP_USER_SHOW` | Show on local  | Show on remote  |
+| :-------------------: | :------------- | :-------------- |
+| `false`               | Never          | Never           |
+| `always`              | Always         | Always          |
+| `true`                | If needed      | Always          |
+| `needed`              | If needed      | If needed       |
+
+### Hostname (`host`)
+
+Hostname is shown only when you're connected via SSH unless you change this behavior.
+
+| Variable | Default | Meaning |
+| :------- | :-----: | ------- |
+| `SPACESHIP_HOST_SHOW` | `true` | Show host section (`true`, `false` or `always`) |
+| `SPACESHIP_HOST_PREFIX` | `at ` | Prefix before the connected SSH machine name |
+| `SPACESHIP_HOST_SUFFIX` | `$SPACESHIP_PROMPT_DEFAULT_SUFFIX` | Suffix after the connected SSH machine name |
+| `SPACESHIP_HOST_COLOR` | `blue` | Color of host section |
+| `SPACESHIP_HOST_COLOR_SSH` | `green` | Color of host in SSH connection |
+
+### Directory (`dir`)
+
+Directory is always shown and truncated to the value of `SPACESHIP_DIR_TRUNC`. While you are in repository, it shows only root directory and folders inside it.
+
+| Variable | Default | Meaning |
+| :------- | :-----: | ------- |
+| `SPACESHIP_DIR_SHOW` | `true` | Show directory section |
+| `SPACESHIP_DIR_PREFIX` | `in ` | Prefix before current directory |
+| `SPACESHIP_DIR_SUFFIX` | `$SPACESHIP_PROMPT_DEFAULT_SUFFIX` | Suffix after current directory |
+| `SPACESHIP_DIR_TRUNC` | `3` | Number of folders of cwd to show in prompt, 0 to show all |
+| `SPACESHIP_DIR_TRUNC_REPO` | `true` | While in `git` repo, show only root directory and folders inside it |
+| `SPACESHIP_DIR_COLOR` | `cyan` | Color of directory section |
+
+### Git (`git`)
+
+Git section is consists with `git_branch` and `git_status` subsections. It is shown only in Git repositories.
+
+| Variable | Default | Meaning |
+| :------- | :-----: | ------- |
+| `SPACESHIP_GIT_SHOW` | `true` | Show Git section |
+| `SPACESHIP_GIT_PREFIX` | `on ` | Prefix before Git section |
```

Comment:
Markdown will strip the leading and trailing whitespaces in inline code blocks, so rendered documentation will be incorrect!

See this for context: https://github.com/denysdovhan/spaceship-zsh-theme/pull/313#discussion_r161926738

Reply:
![image](https://user-images.githubusercontent.com/1177900/35041796-5437a9f0-fb86-11e7-9667-a5725f8f9e58.png)

Reply:
Using NO-BREAK SPACES don't help with GitBook. It still strips trailing space.

Reply:
@maximbaz can you help with that?

Reply:
sure, but what's the plan then? Convert to no-break space and ignore that GitBook would still be incorrect?

Reply:
@maximbaz let's fix this at least for GitHub. Please, keep spaces in code as is, replace only those in docs.

Reply:
all right, I'll do it in an hour, need to shortly leave now.

Reply:
no problem 👌 
waiting for you PR